homework/src/index.js

```diff
@@ -22,25 +23,114 @@
+  const div = createAndAppend('div', root, { class: 'data' });
+  const header = createAndAppend('p', div, { text: 'HYF Repositories', class: 'header' });
+  const select = createAndAppend('select', header, { id: 'list' });
+  const table = createAndAppend('table', root, { class: 'table' });
+  const tbody = createAndAppend('tbody', table);
+  const tr = createAndAppend('tr', tbody, { text: 'Repository: ', class: 'label' });
+
+  const repositoryLink = createAndAppend('a', tr);
+  const description = createAndAppend('tr', tbody, { class: 'label' });
+  const forks = createAndAppend('tr', tbody, { class: 'label' });
+  const container = createAndAppend('div', root, { id: 'container' });
+  const updated = createAndAppend('tr', tbody, { class: 'label' });
+
   function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
-    });
-  }
+    fetchJSON(url)
+      .then(repositories => {
+        repositories.forEach((item, index) => {
+          createAndAppend('option', select, {
+            text: item.name,
+            value: index
+          });
+        });
 
+        repositories.sort((a, b) => a.name.localeCompare(b.name));
+        select.addEventListener('change', () => {
+          container.innerHTML = "";
+          const index = event.target.value;
+
+          renderRepositoryBox(
+            repositories[select.value],
+          );
+          const contributors_url = repositories[index].contributors_url;
+          fetchJSON(contributors_url)
+            .then(contributorData => {
+              renderContributors(contributorData, container);
+            })
+            .catch(error => {
+              createAndAppend('div', container, {
+                text: error.message,
+                class: 'alert-error'
+              });
+              return;
+            });
+
+        });
+
+        const contributorsInfo = repositories[0].contributors_url;
+        fetchJSON(contributorsInfo)
+          .then(contributorData => {
+            renderContributors(contributorData, container);
+          })
+          .catch(error => {
+            createAndAppend('div', container, {
+              text: error.message,
+              class: 'alert-error'
+            });
+            return;
+          });
+
+        function renderRepositoryBox(repositories) {
+          repositoryLink.innerText = repositories.name;
+          repositoryLink.setAttribute('href', repositories.html_url);
+          repositoryLink.setAttribute('target', '_blank');
+          forks.innerHTML = 'Forks: ' + repositories.forks;
+          description.innerHTML = 'Description: ' + repositories.description;
+          updated.innerHTML = 'Updated: ' + new Date(repositories.updated_at).toLocaleString();
```

Comment:
You are creating invalid HTML here. A `tr` element can only have `td` and `th` elements as its direct children.  It should not contain text nodes. But you are creating this:

```html
<table class="table">
  <tbody>
    <tr class="label">Repository: <a href="https://github.com/HackYourFuture/alumni" target="_blank">alumni</a></tr>
    <tr class="label">Description: Alumni Guides, everything you need to land a job</tr>
    <tr class="label">Forks: 8</tr>
    <tr class="label">Updated: 16/07/2018, 11:46:31</tr>
  </tbody>
</table>
```

Reply:
I have changed also this part in JS-WEEK3 and  i will do that aslo in week1 and week2